fleet/fleet.go

```diff
@@ -0,0 +1,87 @@
+// Package fleet implements a fleet client providing basic operations against a
+// fleet endpoint through fleet's HTTP API. Higher level scheduling and
+// management should be built on top of that.
+package fleet
+
+import (
+	"net"
+	"net/http"
+
+	"github.com/coreos/fleet/client"
+	"github.com/coreos/fleet/etcd"
+	"github.com/coreos/fleet/machine"
+	"github.com/coreos/fleet/registry"
+	"github.com/coreos/fleet/schema"
+	"github.com/coreos/fleet/unit"
+)
+
+// TODO
+type Config struct{}
+
+// TODO
+func DefaultConfig() Config {
+	newConfig := Config{}
+
+	return newConfig
+}
+
+type Status struct {
+	// SystemdStatus represents the status within systemd.
+	SystemdStatus containerstates.State
+
+	// FleetStatus represents the status within fleet.
+	FleetStatus containerstates.State
```

Comment:
> Why is it the most interesting one?

LoadState just represents whether the unit is loaded or not. Fleetctl itself showed it in the past, but dropped it because its mostly useless. We are interested in whether things are running, which are represented by ActiveState. SubState is just a more detailed version of ActiveState.

> Where does your FleetCurrent and FleetDesired come from?

from the `Unit` itself, you have the UnitState here.